sandman/source/shell_attr.h

```diff
@@ -0,0 +1,392 @@
+#pragma once
+
+#include "common/box.h"
+#include "common/enum.h"
+
+#include <cstdint>
+#include <curses.h>
+#include <limits>
+#include <utility>
+#include <array>
+#include <string_view>
+
+namespace Shell
+{
+
+	// Represents a bundle of character attributes for the Curses library.
+	struct Attr
```

Comment:
Maybe this should have an unabbreviated name such as `CharacterAttributeBundle` or `AttributeBundle`.